connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -219,7 +230,7 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
                                     log.trace("Wrote record successfully: topic {} partition {} offset {}",
                                             recordMetadata.topic(), recordMetadata.partition(),
                                             recordMetadata.offset());
-                                    commitTaskRecord(record);
+                                    commitTaskRecord(preTransformRecord);
```

Comment:
This is interesting, I don't think we really considered the potentially 2x memory usage increase this can cause.